src/index.ts

```diff
@@ -1,16 +1,48 @@
-const prettier = require('prettier');
-const { RuntimeGlobals } = require('webpack');
+import * as prettier from 'prettier';
+import {Compiler, RuntimeGlobals} from 'webpack';
 
 const pluginName = 'RetryChunkLoadPlugin';
 
-class RetryChunkLoadPlugin {
-  constructor(options = {}) {
+interface Options {
+  /**
+   * optional stringified function to get the cache busting query string appended to the script src
+   * if not set will default to appending the string `?cache-bust=true`
+   */
+  cacheBust?: string;
+  /**
+   * optional list of chunks to which retry script should be injected
+   * if not set will add retry script to all chunks that have webpack script loading
+   */
+  chunks?: string[];
+  /**
+   * optional code to be executed in the browser context if after all retries chunk is not loaded.
+   * if not set - nothing will happen and error will be returned to the chunk loader.
+   */
+  lastResortScript?: string;
+  /**
+   * optional value to set the maximum number of retries to load the chunk. Default is 1
+   */
+  maxRetries?: number;
+  /**
+   * optional value to set the amount of time in milliseconds before trying to load the chunk again. Default is 0
+   */
+  retryDelay?: number;
+  /**
+   * @deprecated - please use `retryDelay` instead
+   */
+  timeout?: number;
+}
+
+export class RetryChunkLoadPlugin {
+  options: Options;
+
+  constructor(options: Options = {}) {
     this.options = Object.assign({}, options);
   }
 
-  apply(compiler) {
-    compiler.hooks.thisCompilation.tap(pluginName, (compilation) => {
-      const { mainTemplate, runtimeTemplate } = compilation;
+  apply(compiler: Compiler) {
+    compiler.hooks.thisCompilation.tap(pluginName, compilation => {
+      const {mainTemplate, runtimeTemplate} = compilation;
```

Comment:
There was a time in my programming career that seeing the bracket spacing change would have upset me, but it's nice that I don't care about pointless stuff like this anymore 😄

Reply:
haha, I tend to offload that decision making to gts defaults, but they are overridable! spaces are back

Reply:
Oh ok, even better then! OCD appeased for another day 😅